packages/plugin-ext/src/main/browser/plugin-shared-style.ts

```diff
@@ -83,26 +93,31 @@ export class PluginSharedStyle {
         }
     }
 
+    private readonly icons = new ReferenceCollection<PluginIconKey, PluginIcon>(key => this.createPluginIcon(key));
```

Comment:
Would it make sense to create a synchronous version of `ReferenceCollection` to avoid the seemingly unnecessary async change?